src/CalcViewModel/StandardCalculatorViewModel.cpp

```diff
@@ -203,6 +204,21 @@ void StandardCalculatorViewModel::SetPrimaryDisplay(_In_ wstring const &displayS
 
     DisplayValue = localizedDisplayStringValue;
 
+    if (!IsFToEAuto && !IsFToEChecked && displayStringValue.find(L"e") != std::wstring::npos)
```

Comment:
I don't feel `displayStringValue.find(L"e")` is a reliable way to test if the current number is in scientific notation, some error messages for example contain a 'e'.

`CCalcEngine::SetPrimaryDisplay` and `CalculatorDisplay::SetPrimaryDisplay` should have an extra parameter indicating the format used.

Reply:
It's not affected by error messages because F-E is disabled while error messages are displayed and it looks nice to have F-E enabled instantly after you type "e" to indicate you are writing number in exponential format. But I'll rewrite to use extra parameter to avoid any troubles with future updates that may introduce character "e" even on non-error states.